File: Books/Queues/PriorityQueue/JS/priorityQueue.js
class PriorityQ {
    #maxSize;
    #nItems;
    #priorityQArray;

    constructor(size) {
        this.#maxSize = size;
        this.#priorityQArray = new Array(this.#maxSize);
        this.#nItems = 0;
    }

    insert(value) {
        let j;

        if (this.#nItems === 0) {
            this.#priorityQArray[this.#nItems++] = value;
        } else {
            for (j = this.#nItems - 1; j >= 0; j--) {
                if (value > this.#priorityQArray[j]) {
                    this.#priorityQArray[j + 1] = this.#priorityQArray[j];
                } else {
                    break;
                }
            }
            this.#priorityQArray[j + 1] = value;
            this.#nItems++;
        }
    }

    remove() {
        return this.#priorityQArray[--this.#nItems];
    }

    peek() {
        return this.#priorityQArray[this.#nItems - 1];
    }

    isEmpty() {
        return this.#nItems === 0;
    }

    isFull() {
        return this.#nItems === this.#maxSize;
    }
}

(function PriorityQApp() {
    const thePQ = new PriorityQ(5);

    thePQ.insert(40);
    thePQ.insert(30);
    thePQ.insert(10);
    thePQ.insert(50);
    thePQ.insert(20);

    while (!thePQ.isEmpty()) {
        const temp = thePQ.remove();
        console.log(temp + " ");
    }

    console.log(" ")
})()

File: Books/JavaRLafore/Queues/Java/PriorityQueue/priorityQueue.java
package Books.Queues.Java.PriorityQueue;

import java.io.IOException;

class PriorityQApp {
    public static void main(String[] args) throws IOException {

        PriorityQ thePQ = new PriorityQ(5);

        thePQ.insert(30);
        thePQ.insert(10);
        thePQ.insert(20);
        thePQ.insert(50);
        thePQ.insert(40);

        while (!thePQ.isEmpty()) {
            long temp = thePQ.remove();

            System.out.print(temp + " ");
        }

        System.out.println(" ");
    }
}

class PriorityQ {
    private int maxSize;
    private long[] priorityQArray;
    private int nItems;

    public PriorityQ(int s) {
        maxSize = s;
        priorityQArray = new long[maxSize];
        nItems = 0;
    }

    public void insert(long item) {
        int j;

        if (nItems == 0) {
            priorityQArray[nItems++] = item;
        } else {
            for (j = nItems - 1; j >= 0; j--) {
                if (item > priorityQArray[j]) {
                    priorityQArray[j + 1] = priorityQArray[j];
                } else {
                    break;
                }
            }

            priorityQArray[j + 1] = item;
            nItems++;
        }
    }

    public long remove() {
        return priorityQArray[--nItems];
    }

    public long peekMin() {
        return priorityQArray[nItems - 1];
    }

    public boolean isEmpty() {
        return nItems == 0;
    }

    public boolean isFull() {
        return nItems == maxSize;
    }
}